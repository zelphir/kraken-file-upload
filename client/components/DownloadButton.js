import React from 'react'
import PropTypes from 'prop-types'
import { AnchorButton } from '@blueprintjs/core'
import { apiUrl } from '../consts'

const DownloadButton = ({ filename, children, ...props }) => {
  return (
    <AnchorButton minimal {...props} href={`${apiUrl}/files/${filename}`}>
      {children}
    </AnchorButton>
  )
}

DownloadButton.propTypes = {
  filename: PropTypes.string.isRequired,
  children: PropTypes.string
}

export default DownloadButton
