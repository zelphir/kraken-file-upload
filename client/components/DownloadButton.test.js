import React from 'react'
import { shallow } from 'enzyme'
import DownloadButton from './DownloadButton'

function setup(nextProps = {}) {
  const props = {
    filename: 'a.txt',
    ...nextProps
  }
  const wrapper = shallow(<DownloadButton {...props} />)

  return {
    props,
    wrapper
  }
}

describe('<DownloadButton />', () => {
  it('should match the snapshot', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the AnchorButton with the icon if icon', () => {
    const { wrapper } = setup({ icon: 'search' })
    expect(wrapper.find('AnchorButton')).toHaveLength(1)
    expect(wrapper.find('AnchorButton').props().icon).toEqual('search')
  })

  it('should render the AnchorButton with a text if children', () => {
    const { wrapper } = setup({ children: 'MyButton' })
    expect(wrapper.find('AnchorButton')).toHaveLength(1)
    expect(wrapper.find('AnchorButton').props().children).toEqual('MyButton')
  })
})
