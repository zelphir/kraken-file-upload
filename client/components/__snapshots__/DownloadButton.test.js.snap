// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownloadButton /> should match the snapshot 1`] = `
<Blueprint2.AnchorButton
  href="http://localhost:5678/files/a.txt"
  minimal={true}
/>
`;
